src/auth/cognito/cognitoAuth.ts

```diff
@@ -0,0 +1,185 @@
+import { AuthInterface } from "../auth";
+import { AuthDevice } from "../types/authDevice";
+import { AuthSession } from "../types/authSession";
+import {
+  AuthCodeDeliveryDetails,
+  AuthStandardAtrributeKey,
+  AuthUserAttribute,
+} from "../types/models";
+import {
+  SignUpRequest,
+  ConfirmSignUpRequest,
+  ResendSignUpCodeRequest,
+  SignInRequest,
+  ConfirmSignInRequest,
+  SignInWithWebUIRequest,
+  FetchAuthSessionRequest,
+  ForgetDeviceRequest,
+  FetchUserAttributesRequest,
+  UpdateUserAttributesRequest,
+  UpdateUserAttributeRequest,
+  ConfirmUserAttributeRequest,
+  ResendUserAttributeConfirmationCodeRequest,
+  SetPreferredMFARequest,
+  UpdatePasswordRequest,
+  ResetPasswordRequest,
+  ConfirmResetPasswordRequest,
+  SignOutRequest,
+  ConfirmSoftwareTokenRequest,
+} from "../types/request";
+import { GetCurrentUserRequest } from "../types/request/getCurrentUserRequest";
+import {
+  AuthSignUpResult,
+  AuthSignInResult,
+  UpdateUserAttributesResult,
+  AuthUser,
+  ResetPasswordResult,
+  SignOutResult,
+  SetupSoftwareTokenResult,
+} from "../types/result";
+import { UpdateUserAttributeResult } from "../types/result/updateUserAttributeResult";
+import { AuthSignInStep } from "../types/step";
+import { CognitoUserAttributeKey, CustomAttribute } from "./types/models";
+import { CognitoMFAType } from "./types/models/cognitoMfaType";
+import {
+  CognitoConfirmSignUpOptions,
+  CognitoSignUpOptions,
+  CognitoResendSignUpCodeOptions,
+  CognitoSignInOptions,
+  CognitoConfirmSignInOptions,
+  CognitoSignInWithWebUIOptions,
+  CognitoUpdateUserAttributesOptions,
+  CognitoUpdateUserAttributeOptions,
+  CognitoResendUserAttributeConfirmationCodeOptions,
+  CognitoResetPasswordOptions,
+} from "./types/options";
+import { TOTPNamespace } from "./types/totpNameSpace";
+
+export class AuthClass implements AuthInterface {
```

Comment:
Can we omit any implementation files (even if they have empty implementations) from the RFC and just focus on the type definitions?